<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><meta name="generator" content="MATLAB R2019b"><title>rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET</title><style type="text/css">.rtcContent { padding: 30px; } .S0 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 28.8px; min-height: 0px; white-space: pre-wrap; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 24px; font-weight: 400; text-align: left;  }
.S1 { margin: 2px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: pre-wrap; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: 400; text-align: left;  }
.S2 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: 700; text-align: left;  }
.S3 { margin: 20px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: 700; text-align: left;  }
.CodeBlock { background-color: #F7F7F7; margin: 10px 0 10px 0;}
.S4 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0px none rgb(0, 0, 0); border-radius: 4px 4px 0px 0px; padding: 6px 45px 0px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S5 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0px none rgb(0, 0, 0); border-radius: 0px; padding: 0px 45px 0px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S6 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 0px; padding: 0px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S7 { color: rgb(64, 64, 64); padding: 10px 0px 6px 17px; background: rgb(255, 255, 255) none repeat scroll 0% 0% / auto padding-box border-box; font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px; overflow-x: hidden; line-height: 16.996px;  }
.embeddedOutputsErrorElement {min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsErrorElement.inlineElement {}
.embeddedOutputsErrorElement.rightPaneElement {}
.embeddedOutputsWarningElement{min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsWarningElement.inlineElement {}
.embeddedOutputsWarningElement.rightPaneElement {}
.diagnosticMessage-wrapper {font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 12px;}
.diagnosticMessage-wrapper.diagnosticMessage-warningType {color: rgb(255,100,0);}
.diagnosticMessage-wrapper.diagnosticMessage-warningType a {color: rgb(255,100,0); text-decoration: underline;}
.diagnosticMessage-wrapper.diagnosticMessage-errorType {color: rgb(230,0,0);}
.diagnosticMessage-wrapper.diagnosticMessage-errorType a {color: rgb(230,0,0); text-decoration: underline;}
.diagnosticMessage-wrapper .diagnosticMessage-messagePart,.diagnosticMessage-wrapper .diagnosticMessage-causePart {white-space: pre-wrap;}
.diagnosticMessage-wrapper .diagnosticMessage-stackPart {white-space: pre;}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement {white-space: pre; word-wrap: initial; min-height: 18px; max-height: 250px; overflow: auto;}
.textElement,.rtcDataTipElement .textElement {padding-top: 3px;}
.embeddedOutputsTextElement.inlineElement,.embeddedOutputsVariableStringElement.inlineElement {}
.inlineElement .textElement {}
.embeddedOutputsTextElement.rightPaneElement,.embeddedOutputsVariableStringElement.rightPaneElement {min-height: 16px;}
.rightPaneElement .textElement {padding-top: 2px; padding-left: 9px;}
.S8 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 0px; padding: 6px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S9 { margin: 10px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: pre-wrap; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: 400; text-align: left;  }
.S10 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 4px 4px 0px 0px; padding: 6px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S11 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0px none rgb(0, 0, 0); border-radius: 0px; padding: 6px 45px 0px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.variableValue { width: 100% !important; }
.embeddedOutputsMatrixElement,.eoOutputWrapper .matrixElement {min-height: 18px; box-sizing: border-box;}
.embeddedOutputsMatrixElement .matrixElement,.eoOutputWrapper .matrixElement,.rtcDataTipElement .matrixElement {position: relative;}
.matrixElement .variableValue,.rtcDataTipElement .matrixElement .variableValue {white-space: pre; display: inline-block; vertical-align: top; overflow: hidden;}
.embeddedOutputsMatrixElement.inlineElement {}
.embeddedOutputsMatrixElement.inlineElement .topHeaderWrapper {display: none;}
.embeddedOutputsMatrixElement.inlineElement .veTable .body {padding-top: 0 !important; max-height: 100px;}
.inlineElement .matrixElement {max-height: 300px;}
.embeddedOutputsMatrixElement.rightPaneElement {}
.rightPaneElement .matrixElement,.rtcDataTipElement .matrixElement {overflow: hidden; padding-left: 9px;}
.rightPaneElement .matrixElement {margin-bottom: -1px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer,.eoOutputWrapper .matrixElement .valueContainer,.rtcDataTipElement .matrixElement .valueContainer {white-space: nowrap; margin-bottom: 3px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis.hide,.embeddedOutputsMatrixElement .matrixElement .verticalEllipsis.hide,.eoOutputWrapper .matrixElement .valueContainer .horizontalEllipsis.hide,.eoOutputWrapper .matrixElement .verticalEllipsis.hide,.rtcDataTipElement .matrixElement .valueContainer .horizontalEllipsis.hide,.rtcDataTipElement .matrixElement .verticalEllipsis.hide {display: none;}
.embeddedOutputsVariableMatrixElement .matrixElement .valueContainer.hideEllipses .verticalEllipsis, .embeddedOutputsVariableMatrixElement .matrixElement .valueContainer.hideEllipses .horizontalEllipsis {display:none;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis,.eoOutputWrapper .matrixElement .valueContainer .horizontalEllipsis {margin-bottom: -3px;}
.eoOutputWrapper .embeddedOutputsVariableMatrixElement .matrixElement .valueContainer {cursor: default !important;}
.embeddedOutputsVariableElement {white-space: pre-wrap; word-wrap: break-word; min-height: 18px; max-height: 250px; overflow: auto;}
.variableElement {}
.embeddedOutputsVariableElement.inlineElement {}
.inlineElement .variableElement {}
.embeddedOutputsVariableElement.rightPaneElement {min-height: 16px;}
.rightPaneElement .variableElement {padding-top: 2px; padding-left: 9px;}
.variableNameElement {margin-bottom: 3px; display: inline-block;}
.matrixElement .horizontalEllipsis,.rtcDataTipElement .matrixElement .horizontalEllipsis {display: inline-block; margin-top: 3px; width: 30px; height: 12px; background-repeat: no-repeat; background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB0AAAAJCAYAAADO1CeCAAAAJUlEQVR42mP4//8/A70xw0i29BUDFPxnAEtTW37wWDqakIa4pQDvOOG89lHX2gAAAABJRU5ErkJggg==");}
.matrixElement .verticalEllipsis,.textElement .verticalEllipsis,.rtcDataTipElement .matrixElement .verticalEllipsis,.rtcDataTipElement .textElement .verticalEllipsis {margin-left: 35px; width: 12px; height: 30px; background-repeat: no-repeat; background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAZCAYAAAAIcL+IAAAALklEQVR42mP4//8/AzGYgWyFMECMwv8QddRS+P//KyimlmcGUOFoOI6GI/UVAgDnd8Dd4+NCwgAAAABJRU5ErkJggg==");}
.S12 { margin: 15px 10px 5px 4px; padding: 0px; line-height: 18px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 17px; font-weight: 700; text-align: left;  }
.S13 { margin: 10px 0px 20px; padding-left: 0px; font-family: Helvetica, Arial, sans-serif; font-size: 14px;  }
.S14 { margin-left: 56px; line-height: 21px; min-height: 0px; text-align: left; white-space: pre-wrap;  }
.S15 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 0px 0px 4px 4px; padding: 0px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S16 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 18px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 17px; font-weight: 700; text-align: left;  }
.S17 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 4px; padding: 6px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }</style></head><body><div class = rtcContent><h1  class = 'S0'><span>rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET</span></h1><div  class = 'S1'><span>(Pacheco et al., 2019, EBioMedicine)</span></div><h2  class = 'S2'><span></span></h2><h2  class = 'S3'><span>SET UP</span></h2><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>addpath(genpath(pwd)) </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>addpath(genpath(</span><span style="color: rgb(160, 32, 240);">'/Users/claudiavicentecomorera/cobratoolbox'</span><span>)) </span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S6'><span style="white-space: pre;"><span>feature </span><span style="color: rgb(160, 32, 240);">astheightlimit 2000</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableStringElement" uid="9C947231" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="21" data-hashorizontaloverflow="false" data-testid="output_0" style="max-height: 261px; width: 762.026px;"><div class="textElement"><div><span class="variableNameElement">ans = </span>'feature astHeightLimit 500'</div></div></div></div></div><div class="inlineWrapper outputs"><div  class = 'S8'><span style="white-space: pre;"><span>initCobraToolbox(false)</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement scrollableOutput" uid="47EB5A20" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="2432" data-hashorizontaloverflow="true" data-testid="output_1" style="max-height: 261px; width: 762.026px;"><div class="textElement">      _____   _____   _____   _____     _____     |
     /  ___| /  _  \ |  _  \ |  _  \   / ___ \    |   COnstraint-Based Reconstruction and Analysis
     | |     | | | | | |_| | | |_| |  | |___| |   |   The COBRA Toolbox - 2024
     | |     | | | | |  _  { |  _  /  |  ___  |   |
     | |___  | |_| | | |_| | | | \ \  | |   | |   |   Documentation:
     \_____| \_____/ |_____/ |_|  \_\ |_|   |_|   |   <a href="http://opencobra.github.io/cobratoolbox">http://opencobra.github.io/cobratoolbox</a>
                                                  | 

 &gt; Checking if git is installed ...  Done (version: 2.39.3).
 &gt; Checking if the repository is tracked using git ...  Done.
 &gt; Checking if curl is installed ...  Done.
 &gt; Checking if remote can be reached ...  Done.
 &gt; Initializing and updating submodules (this may take a while)... Done.
 &gt; Adding all the files of The COBRA Toolbox ...  Done.
 &gt; Define CB map output... set to svg.
 &gt; TranslateSBML is installed and working properly.
 &gt; Configuring solver environment variables ...
   - [*---] ILOG_CPLEX_PATH: /Applications/CPLEX_Studio1210/cplex/matlab/x86-64_osx
   - [----] GUROBI_PATH: --&gt; set this path manually after installing the solver ( see <a href="https://opencobra.github.io/cobratoolbox/docs/solvers.html">instructions</a> )
   - [----] TOMLAB_PATH: --&gt; set this path manually after installing the solver ( see <a href="https://opencobra.github.io/cobratoolbox/docs/solvers.html">instructions</a> )
   - [----] MOSEK_PATH: --&gt; set this path manually after installing the solver ( see <a href="https://opencobra.github.io/cobratoolbox/docs/solvers.html">instructions</a> )
   Done.
 &gt; Checking available solvers and solver interfaces ...Gurobi installed at this location? 
Licence file current? 
Version identifier: 12.10.0.0 | 2019-11-26 | 843d4de
CPXPARAM_Output_WriteLevel                       3
CPXPARAM_Output_CloneLog                         -1
Found incumbent of value 0.000000 after 0.00 sec. (0.00 ticks)

Root node processing (before b&amp;c):
  Real time             =    0.01 sec. (0.00 ticks)
Parallel b&amp;c, 8 threads:
  Real time             =    0.00 sec. (0.00 ticks)
  Sync time (average)   =    0.00 sec.
  Wait time (average)   =    0.00 sec.
                          ------------
Total (root+branch&amp;cut) =    0.01 sec. (0.00 ticks)
Could not find installation of tomlab_cplex, so it cannot be tested
Original LP has 1 row, 2 columns, 1 non-zero
Objective value = 0
OPTIMAL SOLUTION FOUND BY LP PRESOLVER

 &gt; [glpk] Primal optimality condition in solveCobraLP satisfied.Could not find installation of mosek, so it cannot be tested

   --------------------------------------------------------
   pdco.m                      Version pdco5 of 15 Jun 2018
   Primal-dual barrier method to minimize a convex function
   subject to linear constraints Ax + r = b,  bl &lt;= x &lt;= bu
                                                           
   Michael Saunders       SOL and ICME, Stanford University
   Contributors:     Byunggyoo Kim (SOL), Chris Maes (ICME)
                     Santiago Akle (ICME), Matt Zahr (ICME)
                     Aekaansh Verma (ME)                   
   --------------------------------------------------------

The objective is linear
The matrix A is an explicit sparse matrix

m        =        1     n        =        2      nnz(A)  =        1
max |b | =        0     max |x0| =  1.0e+00      xsize   =  1.0e+00
max |y0| =        1     max |z0| =  1.0e+00      zsize   =  1.0e+00

x0min    =        1     featol   =  1.0e-06      d1max   =  1.0e-04
z0min    =        1     opttol   =  1.0e-06      d2max   =  5.0e-04
mu0      =  1.0e-01     steptol  =     0.99     bigcenter=     1000

LSMR/MINRES:
atol1    =  1.0e-10     atol2    =  1.0e-15      btol    =  0.0e+00
conlim   =  1.0e+12     itnlim   =       10      show    =        0

Method   =        2     (1 or 11=chol  2 or 12=QR  3 or 13=LSMR  4 or 14=MINRES 21=SQD(LU)  22=SQD(MA57))
Eliminating dy before dx  
 

Bounds:
  [0,inf]  [-inf,0]  Finite bl  Finite bu  Two bnds   Fixed    Free
        0         0          0          0         0       2       0
  [0, bu]  [bl,  0]  excluding fixed variables
        0         0

Itn   mu stepx stepz  Pinf  Dinf  Cinf   Objective    nf  center       QR
  0                   -6.6 -99.0  -Inf  1.2500000e-07        1.0
  1 -1.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0        1
  2 -3.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
  3 -5.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
  4 -7.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
   Converged

max |x| =     0.000    max |y| =     0.000    max |z| =     0.000   scaled
max |x| =     0.000    max |y| =     0.000    max |z| =     0.000 unscaled
max |x| and max |z| exclude fixed variables
PDitns  =         4     QRitns =         0    cputime =       0.2

Distribution of vector     x         z
[      1,     10 )         0         2
[    0.1,      1 )         0         0
[   0.01,    0.1 )         0         0
[  0.001,   0.01 )         0         0
[ 0.0001,  0.001 )         0         0
[  1e-05, 0.0001 )         0         0
[  1e-06,  1e-05 )         0         0
[  1e-07,  1e-06 )         0         0
[  1e-08,  1e-07 )         0         0
[      0,  1e-08 )         2         0 
Elapsed time is 0.222692 seconds.

 &gt; [pdco] Primal optimality condition in solveCobraLP satisfied.
 &gt; [pdco] Dual optimality condition in solveCobraLP satisfied.
Elapsed time is 0.003358 seconds.
In 0.0068551 sec, file written to /Users/claudiavicentecomorera/cobratoolbox/binary/maci64/bin/minos/data/FBA/qFBA.txt
Could not find installation of cplex_direct, so it cannot be tested

 &gt; [cplexlp] Primal optimality condition in solveCobraLP satisfied.
 &gt; [cplexlp] Dual optimality condition in solveCobraLP satisfied.
Could not find installation of tomlab_snopt, so it cannot be tested
 Done.
 &gt; Setting default solvers ...Could not find installation of mosek, so it cannot be tested
 Done.
 &gt; Saving the MATLAB path ... Done.
   - The MATLAB path was saved in the default location.

 &gt; Summary of available solvers and solver interfaces

			Support 	   LP 	 MILP 	   QP 	 MIQP 	  NLP 	   EP
	------------------------------------------------------------------------------
	gurobi       	active        	    0 	    0 	    0 	    0 	    - 	    -
	ibm_cplex    	active        	    1 	    1 	    1 	    1 	    - 	    -
	tomlab_cplex 	active        	    0 	    0 	    0 	    0 	    - 	    -
	glpk         	active        	    1 	    1 	    - 	    - 	    - 	    -
	mosek        	active        	    0 	    - 	    0 	    - 	    - 	    0
	matlab       	active        	    1 	    - 	    - 	    - 	    1 	    -
	pdco         	active        	    1 	    - 	    1 	    - 	    - 	    1
	quadMinos    	active        	    1 	    - 	    - 	    - 	    - 	    -
	dqqMinos     	active        	    1 	    - 	    1 	    - 	    - 	    -
	cplex_direct 	active        	    0 	    0 	    0 	    - 	    - 	    -
	cplexlp      	active        	    1 	    - 	    - 	    - 	    - 	    -
	qpng         	passive       	    - 	    - 	    1 	    - 	    - 	    -
	tomlab_snopt 	passive       	    - 	    - 	    - 	    - 	    0 	    -
	lp_solve     	legacy        	    1 	    - 	    - 	    - 	    - 	    -
	------------------------------------------------------------------------------
	Total        	-             	    8 	    2 	    4 	    1 	    1 	    1

 + Legend: - = not applicable, 0 = solver not compatible or not installed, 1 = solver installed.


 &gt; You can solve LP problems using: 'ibm_cplex' - 'glpk' - 'pdco' - 'cplexlp' 
 &gt; You can solve MILP problems using: 'ibm_cplex' - 'glpk' 
 &gt; You can solve QP problems using: 'ibm_cplex' - 'pdco' 
 &gt; You can solve MIQP problems using: 'ibm_cplex' 
 &gt; You can solve NLP problems using: 
 &gt; You can solve EP problems using: 'pdco' 

Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
&gt; Checking for available updates ... skipped
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/componentContribution/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/groupContribution/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/inchi/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/molFiles/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/protons/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/trainingModel/new</div></div></div></div></div><div  class = 'S9'><span>Import the data and split into 3 variables:</span></div><div class="CodeBlock"><div class="inlineWrapper outputs"><div  class = 'S10'><span style="white-space: pre;"><span>data = readtable(</span><span style="color: rgb(160, 32, 240);">'data_pbmc.csv'</span><span>);</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsWarningElement" uid="EFC30917" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="45" data-hashorizontaloverflow="false" data-testid="output_2" style="max-height: 261px; width: 762.026px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: Column headers from the file were modified to make them valid MATLAB identifiers before creating variable names for the table. The original column headers are saved in the VariableDescriptions property.<br>Set 'PreserveVariableNames' to true to use the original column headers as table variable names.</div><div class="diagnosticMessage-stackPart"></div></div></div></div></div><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: pre;"><span>colnames = data.Properties.VariableNames(2:end);</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>rownames = table2cell(data(:, 1));</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S6'><span style="white-space: pre;"><span>CPM = table2array(data(:, 2:end))</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableMatrixElement" uid="60028C66" data-scroll-top="null" data-scroll-left="null" data-width="732" data-testid="output_3" style="width: 762.026px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="matrixElement veSpecifier saveLoad" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="veVariableName variableNameElement double" style="width: 732px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="headerElementClickToInteract" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">CPM = </span><span class="veVariableValueSummary veMetaSummary" style="white-space: normal; font-style: normal; color: rgb(179, 179, 179); font-size: 12px;">32738×2638</span></div></div><div class="veScalingFactor" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">10<sup>5</sup><span class="multiply" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"> ×</span></div><div class="valueContainer" data-layout="{&quot;columnWidth&quot;:73,&quot;totalColumns&quot;:&quot;2638&quot;,&quot;totalRows&quot;:&quot;32738&quot;,&quot;charsPerColumn&quot;:10}" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="variableValue" style="width: 659px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
</div><div class="horizontalEllipsis" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div><div class="verticalEllipsis" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div></div></div></div></div></div></div><div  class = 'S9'><span></span></div><div  class = 'S1'><span></span></div><h2  class = 'S3'><span>DISCRETIZATION STEP</span></h2><div  class = 'S1'><span>Discretize the gene expression values into expressed, not expressed and unknown expression status:</span></div><div class="CodeBlock"><div class="inlineWrapper outputs"><div  class = 'S10'><span style="white-space: pre;"><span>discretized = discretize_FPKM(CPM, colnames) </span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableMatrixElement" uid="51504B04" data-scroll-top="null" data-scroll-left="null" data-width="732" data-testid="output_4" style="width: 762.026px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="matrixElement veSpecifier saveLoad" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="veVariableName variableNameElement double" style="width: 732px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="headerElementClickToInteract" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">discretized = </span><span class="veVariableValueSummary veMetaSummary" style="white-space: normal; font-style: normal; color: rgb(179, 179, 179); font-size: 12px;">32738×2638</span></div></div><div class="valueContainer" data-layout="{&quot;columnWidth&quot;:44,&quot;totalColumns&quot;:&quot;2638&quot;,&quot;totalRows&quot;:&quot;32738&quot;,&quot;charsPerColumn&quot;:6}" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="variableValue" style="width: 662px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
</div><div class="horizontalEllipsis" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div><div class="verticalEllipsis" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div></div></div></div></div></div></div><h2  class = 'S2'><span></span></h2><h2  class = 'S3'><span>rFASTCORMICS</span></h2><h3  class = 'S12'><span>NEEDED INPUTS</span></h3><div  class = 'S1'><span>Apart from the data, rFASTCORMICS needs some more inputs before run:</span></div><ul  class = 'S13'><li  class = 'S14'><span>A file which links the geneIDs from the RNA-seq data to the gene identifiers in the model (Entrez).</span></li><li  class = 'S14'><span>A consistent reconstruction (as Recon3D)</span></li></ul><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>load </span><span style="color: rgb(160, 32, 240);">dico_rFASTCORMICS.mat </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>load </span><span style="color: rgb(160, 32, 240);">Recon3DModel_301.mat</span><span>;</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S6'><span style="white-space: pre;"><span>A = fastcc_4_rfastcormics(model, 1e-4,0) </span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="FC385C12" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="32" data-hashorizontaloverflow="false" data-testid="output_5" style="max-height: 261px; width: 762.026px;"><div class="textElement">The input model is consistent.
Elapsed time is 14.364739 seconds.</div></div><div class="inlineElement eoOutputWrapper embeddedOutputsVariableMatrixElement" uid="4971E9F0" data-scroll-top="null" data-scroll-left="null" data-width="732" data-testid="output_6" style="width: 762.026px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="matrixElement veSpecifier saveLoad" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="veVariableName variableNameElement double" style="width: 732px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="headerElementClickToInteract" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">A = </span><span class="veVariableValueSummary veMetaSummary" style="white-space: normal; font-style: normal; color: rgb(179, 179, 179); font-size: 12px;">10600×1</span></div></div><div class="valueContainer" data-layout="{&quot;columnWidth&quot;:44,&quot;totalColumns&quot;:&quot;1&quot;,&quot;totalRows&quot;:&quot;10600&quot;,&quot;charsPerColumn&quot;:6}" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="variableValue" style="width: 46px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">     1
     2
     3
     4
     5
     6
     7
     8
     9
    10
</div><div class="horizontalEllipsis hide" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div><div class="verticalEllipsis" style="white-space: nowrap; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"></div></div></div></div></div></div><div class="inlineWrapper outputs"><div  class = 'S8'><span style="white-space: pre;"><span>Cmodel = removeRxns(model, model.rxns(setdiff(1:numel(model.rxns),A)))</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableStringElement scrollableOutput" uid="025921CB" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="574" data-hashorizontaloverflow="false" data-testid="output_7" style="max-height: 261px; width: 762.026px;"><div class="textElement"><div><span class="variableNameElement">Cmodel = <span class="headerElement">struct with fields:</span></span></div><div>                      S: [5835×10600 double]
                   mets: {5835×1 cell}
                      b: [5835×1 double]
                 csense: [5835×1 char]
                   rxns: {10600×1 cell}
                     lb: [10600×1 double]
                     ub: [10600×1 double]
                      c: [10600×1 double]
                 osense: -1
                  genes: {2248×1 cell}
                  rules: {10600×1 cell}
             metCharges: [5835×1 int64]
            metFormulas: {5835×1 cell}
              metSmiles: {5835×1 cell}
               metNames: {5835×1 cell}
              metHMDBID: {5835×1 cell}
         metInChIString: {5835×1 cell}
              metKEGGID: {5835×1 cell}
           metPubChemID: {5835×1 cell}
            description: 'Recon3DModel.mat'
                grRules: {10600×1 cell}
             rxnGeneMat: [10600×2248 double]
    rxnConfidenceScores: [10600×1 double]
               rxnNames: {10600×1 cell}
               rxnNotes: {10600×1 cell}
           rxnECNumbers: {10600×1 cell}
          rxnReferences: {10600×1 cell}
              rxnKEGGID: {10600×1 cell}
             subSystems: {10600×1 cell}
             metCHEBIID: {5835×1 cell}
               metPdMap: {5835×1 cell}
            metReconMap: {5835×1 cell}
                modelID: 'Recon3DModel'
                 rxnCOG: {10600×1 cell}
       rxnKeggOrthology: {10600×1 cell}
            rxnReconMap: {10600×1 cell}
                version: 'Recon3D_01'
             PleaseCite: 'Brunk et al, Nat Biotech, 2018; doi:10.1038/nbt.4072'
</div></div></div></div></div></div><div  class = 'S9'><span></span></div><div  class = 'S1'><span></span></div><h3  class = 'S12'><span>PARAMETERS AND OPTIONAL INPUTS</span></h3><div  class = 'S1'><span>We will use the parameters and inputs defined in rFASTCORMICS example scripts, excluding the medium constraint.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>biomass_rxn = {</span><span style="color: rgb(160, 32, 240);">'biomass_maintenance'</span><span>}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>already_mapped_tag = 0;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>epsilon = 1e-4; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>consensus_proportion = 0.9; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>unpenalizedSystems = {</span><span style="color: rgb(160, 32, 240);">'Transport, endoplasmic reticular'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, extracellular'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, golgi apparatus'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, mitochondrial'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, peroxisomal'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, lysosomal'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, nuclear'</span><span>};</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>Cmodel_subSystems = cellfun(@(x) x{1}, Cmodel.subSystems, </span><span style="color: rgb(160, 32, 240);">'UniformOutput'</span><span>, false); </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>unpenalized = Cmodel.rxns(ismember(Cmodel_subSystems,unpenalizedSystems));</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>optional_settings.unpenalized = unpenalized; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>optional_settings.func = {</span><span style="color: rgb(160, 32, 240);">'biomass_maintenance'</span><span>,</span><span style="color: rgb(160, 32, 240);">'DM_atp_c_'</span><span>}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>not_medium_constrained = </span><span style="color: rgb(160, 32, 240);">'EX_tag_hs[e]'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S15'><span style="white-space: pre;"><span>optional_settings.not_medium_constrained = not_medium_constrained;</span></span></div></div></div><h3  class = 'S16'><span></span></h3><h3  class = 'S12'><span>CONTEXT-SPECIFIC SINGLE MODELS CREATION</span></h3><div  class = 'S1'><span>By setting all the inputs into the </span><span style=' font-weight: bold; font-style: italic;'>fastcormics_RNAseq</span><span> function, a model for each cell (column) of the data will be build:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span style="color: rgb(0, 0, 255);">for </span><span>i = 1:numel(colnames) </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    [ContextModel, A_keep] = fastcormics_RNAseq(Cmodel, discretized(:,i), </span><span style="color: rgb(0, 0, 255);">...</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>        rownames, dico, biomass_rxn, already_mapped_tag, consensus_proportion, epsilon, optional_settings);</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>     </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    filePath = fullfile(</span><span style="color: rgb(160, 32, 240);">'PBMC_Models'</span><span>, colnames{i});</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    save(filePath, </span><span style="color: rgb(160, 32, 240);">'ContextModel'</span><span>);   </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span class="warning_squiggle_rte">models_keep</span><span>(A_keep,i) = 1;</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S6'><span style="white-space: pre;"><span style="color: rgb(0, 0, 255);">end</span></span></div><div  class = 'S7'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="1E695435" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="46" data-hashorizontaloverflow="false" data-testid="output_8" style="max-height: 261px; width: 762.026px;"><div class="textElement">creating model.rev
unnesting subsystems
2143 of 2248 genes matched</div></div><div class="inlineElement eoOutputWrapper embeddedOutputsWarningElement" uid="157C7F1B" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="18" data-hashorizontaloverflow="false" data-testid="output_9" style="max-height: 261px; width: 762.026px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No optional settings detected</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="5D7CB56D" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="18" data-hashorizontaloverflow="false" data-testid="output_10" style="max-height: 261px; width: 762.026px;"><div class="textElement">Elapsed time is 46.868621 seconds.</div></div><div class="inlineElement eoOutputWrapper embeddedOutputsErrorElement" uid="B743E524" data-scroll-top="null" data-scroll-left="null" data-width="732" data-height="31" data-hashorizontaloverflow="false" data-testid="output_11" style="max-height: 261px; width: 762.026px;"><div class="diagnosticMessage-wrapper diagnosticMessage-errorType"><div class="diagnosticMessage-messagePart">Error using save<br>Cannot create 'AAACATACAACCAC_1.mat' because 'PBMC_Models' does not exist.</div><div class="diagnosticMessage-stackPart"></div></div></div></div></div></div><div  class = 'S9'><span></span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S17'><span style="white-space: pre;"><span>delete </span><span style="color: rgb(160, 32, 240);">clone*.log</span></span></div></div></div></div>
<br>
<!-- 
##### SOURCE BEGIN #####
%% rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET
% (Pacheco et al., 2019, EBioMedicine)
%% 
%% SET UP

addpath(genpath(pwd)) 
addpath(genpath('/Users/claudiavicentecomorera/cobratoolbox')) 
feature astheightlimit 2000
initCobraToolbox(false)
%% 
% Import the data and split into 3 variables:

data = readtable('data_pbmc.csv');
colnames = data.Properties.VariableNames(2:end);
rownames = table2cell(data(:, 1));
CPM = table2array(data(:, 2:end))
%% 
% 
% 
% 
%% DISCRETIZATION STEP
% Discretize the gene expression values into expressed, not expressed and unknown 
% expression status:

discretized = discretize_FPKM(CPM, colnames) 
%% 
%% rFASTCORMICS
% NEEDED INPUTS
% Apart from the data, rFASTCORMICS needs some more inputs before run:
%% 
% * A file which links the geneIDs from the RNA-seq data to the gene identifiers 
% in the model (Entrez).
% * A consistent reconstruction (as Recon3D)

load dico_rFASTCORMICS.mat 
load Recon3DModel_301.mat;
A = fastcc_4_rfastcormics(model, 1e-4,0) 
Cmodel = removeRxns(model, model.rxns(setdiff(1:numel(model.rxns),A)))
%% 
% 
% 
% 
% PARAMETERS AND OPTIONAL INPUTS
% We will use the parameters and inputs defined in rFASTCORMICS example scripts, 
% excluding the medium constraint.

biomass_rxn = {'biomass_maintenance'}; 
already_mapped_tag = 0;
epsilon = 1e-4; 
consensus_proportion = 0.9; 
unpenalizedSystems = {'Transport, endoplasmic reticular';
    'Transport, extracellular';
    'Transport, golgi apparatus';
    'Transport, mitochondrial';
    'Transport, peroxisomal';
    'Transport, lysosomal';
    'Transport, nuclear'};
Cmodel_subSystems = cellfun(@(x) x{1}, Cmodel.subSystems, 'UniformOutput', false); 
unpenalized = Cmodel.rxns(ismember(Cmodel_subSystems,unpenalizedSystems));

optional_settings.unpenalized = unpenalized; 
optional_settings.func = {'biomass_maintenance','DM_atp_c_'}; 
not_medium_constrained = 'EX_tag_hs[e]';
optional_settings.not_medium_constrained = not_medium_constrained;
% 
% CONTEXT-SPECIFIC SINGLE MODELS CREATION
% By setting all the inputs into the _*fastcormics_RNAseq*_ function, a model 
% for each cell (column) of the data will be build:

for i = 1:numel(colnames) 
    [ContextModel, A_keep] = fastcormics_RNAseq(Cmodel, discretized(:,i), ...
        rownames, dico, biomass_rxn, already_mapped_tag, consensus_proportion, epsilon, optional_settings);
     
    filePath = fullfile('PBMC_Models', colnames{i});
    save(filePath, 'ContextModel');   
    models_keep(A_keep,i) = 1;
end
%% 
% 

delete clone*.log
##### SOURCE END #####
--></body></html>