<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><meta name="generator" content="MATLAB R2019b"><title>rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET</title><style type="text/css">.rtcContent { padding: 30px; } .S0 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 28.8px; min-height: 0px; white-space: pre-wrap; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 24px; font-weight: 400; text-align: left;  }
.S1 { margin: 2px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: pre-wrap; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: 400; text-align: left;  }
.S2 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: 700; text-align: left;  }
.S3 { margin: 20px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: 700; text-align: left;  }
.CodeBlock { background-color: #F7F7F7; margin: 10px 0 10px 0;}
.S4 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0px none rgb(0, 0, 0); border-radius: 4px 4px 0px 0px; padding: 6px 45px 0px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S5 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0px none rgb(0, 0, 0); border-radius: 0px; padding: 0px 45px 0px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S6 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 0px 0px 4px 4px; padding: 0px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S7 { margin: 10px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: pre-wrap; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: 400; text-align: left;  }
.S8 { border-left: 0.994318px solid rgb(233, 233, 233); border-right: 0.994318px solid rgb(233, 233, 233); border-top: 0.994318px solid rgb(233, 233, 233); border-bottom: 0.994318px solid rgb(233, 233, 233); border-radius: 4px; padding: 6px 45px 4px 13px; line-height: 16.996px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S9 { margin: 15px 10px 5px 4px; padding: 0px; line-height: 18px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 17px; font-weight: 700; text-align: left;  }
.S10 { margin: 10px 0px 20px; padding-left: 0px; font-family: Helvetica, Arial, sans-serif; font-size: 14px;  }
.S11 { margin-left: 56px; line-height: 21px; min-height: 0px; text-align: left; white-space: pre-wrap;  }
.S12 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 18px; min-height: 0px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 17px; font-weight: 700; text-align: left;  }</style></head><body><div class = rtcContent><h1  class = 'S0'><span>rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET</span></h1><div  class = 'S1'><span>(Pacheco et al., 2019, EBioMedicine)</span></div><h2  class = 'S2'><span></span></h2><h2  class = 'S3'><span>SET UP</span></h2><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>addpath(genpath(pwd)) </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>addpath(genpath(</span><span style="color: rgb(160, 32, 240);">'/Users/claudiavicentecomorera/cobratoolbox'</span><span>)) </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>feature </span><span style="color: rgb(160, 32, 240);">astheightlimit 2000</span></span></div></div><div class="inlineWrapper"><div  class = 'S6'><span style="white-space: pre;"><span>initCobraToolbox(false)</span></span></div></div></div><div  class = 'S7'><span>Import the data and split into 3 variables:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>data = readtable(</span><span style="color: rgb(160, 32, 240);">'data_pbmc.csv'</span><span>);</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>colnames = data.Properties.VariableNames(2:end);</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>rownames = table2cell(data(:, 1));</span></span></div></div><div class="inlineWrapper"><div  class = 'S6'><span style="white-space: pre;"><span>CPM = table2array(data(:, 2:end))</span></span></div></div></div><div  class = 'S7'><span></span></div><div  class = 'S1'><span></span></div><h2  class = 'S3'><span>DISCRETIZATION STEP</span></h2><div  class = 'S1'><span>Discretize the gene expression values into expressed, not expressed and unknown expression status:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: pre;"><span>discretized = discretize_FPKM(CPM, colnames) </span></span></div></div></div><h2  class = 'S2'><span></span></h2><h2  class = 'S3'><span>rFASTCORMICS</span></h2><h3  class = 'S9'><span>NEEDED INPUTS</span></h3><div  class = 'S1'><span>Apart from the data, rFASTCORMICS needs some more inputs before run:</span></div><ul  class = 'S10'><li  class = 'S11'><span>A file which links the geneIDs from the RNA-seq data to the gene identifiers in the model (Entrez).</span></li><li  class = 'S11'><span>A consistent reconstruction (as Recon3D)</span></li></ul><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>load </span><span style="color: rgb(160, 32, 240);">dico_rFASTCORMICS.mat </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>load </span><span style="color: rgb(160, 32, 240);">Recon3DModel_301.mat</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>A = fastcc_4_rfastcormics(model, 1e-4,0) </span></span></div></div><div class="inlineWrapper"><div  class = 'S6'><span style="white-space: pre;"><span>Cmodel = removeRxns(model, model.rxns(setdiff(1:numel(model.rxns),A)))</span></span></div></div></div><div  class = 'S7'><span></span></div><div  class = 'S1'><span></span></div><h3  class = 'S9'><span>PARAMETERS AND OPTIONAL INPUTS</span></h3><div  class = 'S1'><span>We will use the parameters and inputs defined in rFASTCORMICS example scripts, excluding the medium constraint.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span>biomass_rxn = {</span><span style="color: rgb(160, 32, 240);">'biomass_maintenance'</span><span>}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>already_mapped_tag = 0;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>epsilon = 1e-4; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>consensus_proportion = 0.9; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>unpenalizedSystems = {</span><span style="color: rgb(160, 32, 240);">'Transport, endoplasmic reticular'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, extracellular'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, golgi apparatus'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, mitochondrial'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, peroxisomal'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, lysosomal'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span style="color: rgb(160, 32, 240);">'Transport, nuclear'</span><span>};</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>Cmodel_subSystems = cellfun(@(x) x{1}, Cmodel.subSystems, </span><span style="color: rgb(160, 32, 240);">'UniformOutput'</span><span>, false); </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>unpenalized = Cmodel.rxns(ismember(Cmodel_subSystems,unpenalizedSystems));</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>optional_settings.unpenalized = unpenalized; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>optional_settings.func = {</span><span style="color: rgb(160, 32, 240);">'biomass_maintenance'</span><span>,</span><span style="color: rgb(160, 32, 240);">'DM_atp_c_'</span><span>}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>not_medium_constrained = </span><span style="color: rgb(160, 32, 240);">'EX_tag_hs[e]'</span><span>;</span></span></div></div><div class="inlineWrapper"><div  class = 'S6'><span style="white-space: pre;"><span>optional_settings.not_medium_constrained = not_medium_constrained;</span></span></div></div></div><h3  class = 'S12'><span></span></h3><h3  class = 'S9'><span>CONTEXT-SPECIFIC SINGLE MODELS CREATION</span></h3><div  class = 'S1'><span>By setting all the inputs into the </span><span style=' font-weight: bold; font-style: italic;'>fastcormics_RNAseq</span><span> function, a model for each cell (column) of the data will be build:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S4'><span style="white-space: pre;"><span style="color: rgb(0, 0, 255);">for </span><span>i = 1:numel(colnames) </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    [ContextModel, A_keep] = fastcormics_RNAseq(Cmodel, discretized(:,i), </span><span style="color: rgb(0, 0, 255);">...</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>        rownames, dico, biomass_rxn, already_mapped_tag, consensus_proportion, epsilon, optional_settings);</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>     </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    filePath = fullfile(</span><span style="color: rgb(160, 32, 240);">'PBMC_Models'</span><span>, colnames{i});</span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    save(filePath, </span><span style="color: rgb(160, 32, 240);">'ContextModel'</span><span>);   </span></span></div></div><div class="inlineWrapper"><div  class = 'S5'><span style="white-space: pre;"><span>    </span><span class="warning_squiggle_rte">models_keep</span><span>(A_keep,i) = 1;</span></span></div></div><div class="inlineWrapper"><div  class = 'S6'><span style="white-space: pre;"><span style="color: rgb(0, 0, 255);">end</span></span></div></div></div><div  class = 'S7'><span></span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: pre;"><span>delete </span><span style="color: rgb(160, 32, 240);">clone*.log</span></span></div></div></div></div>
<br>
<!-- 
##### SOURCE BEGIN #####
%% rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET
% (Pacheco et al., 2019, EBioMedicine)
%% 
%% SET UP

addpath(genpath(pwd)) 
addpath(genpath('/Users/claudiavicentecomorera/cobratoolbox')) 
feature astheightlimit 2000
initCobraToolbox(false)
%% 
% Import the data and split into 3 variables:

data = readtable('data_pbmc.csv');
colnames = data.Properties.VariableNames(2:end);
rownames = table2cell(data(:, 1));
CPM = table2array(data(:, 2:end))
%% 
% 
% 
% 
%% DISCRETIZATION STEP
% Discretize the gene expression values into expressed, not expressed and unknown 
% expression status:

discretized = discretize_FPKM(CPM, colnames) 
%% 
%% rFASTCORMICS
% NEEDED INPUTS
% Apart from the data, rFASTCORMICS needs some more inputs before run:
%% 
% * A file which links the geneIDs from the RNA-seq data to the gene identifiers 
% in the model (Entrez).
% * A consistent reconstruction (as Recon3D)

load dico_rFASTCORMICS.mat 
load Recon3DModel_301.mat;
A = fastcc_4_rfastcormics(model, 1e-4,0) 
Cmodel = removeRxns(model, model.rxns(setdiff(1:numel(model.rxns),A)))
%% 
% 
% 
% 
% PARAMETERS AND OPTIONAL INPUTS
% We will use the parameters and inputs defined in rFASTCORMICS example scripts, 
% excluding the medium constraint.

biomass_rxn = {'biomass_maintenance'}; 
already_mapped_tag = 0;
epsilon = 1e-4; 
consensus_proportion = 0.9; 
unpenalizedSystems = {'Transport, endoplasmic reticular';
    'Transport, extracellular';
    'Transport, golgi apparatus';
    'Transport, mitochondrial';
    'Transport, peroxisomal';
    'Transport, lysosomal';
    'Transport, nuclear'};
Cmodel_subSystems = cellfun(@(x) x{1}, Cmodel.subSystems, 'UniformOutput', false); 
unpenalized = Cmodel.rxns(ismember(Cmodel_subSystems,unpenalizedSystems));

optional_settings.unpenalized = unpenalized; 
optional_settings.func = {'biomass_maintenance','DM_atp_c_'}; 
not_medium_constrained = 'EX_tag_hs[e]';
optional_settings.not_medium_constrained = not_medium_constrained;
% 
% CONTEXT-SPECIFIC SINGLE MODELS CREATION
% By setting all the inputs into the _*fastcormics_RNAseq*_ function, a model 
% for each cell (column) of the data will be build:

for i = 1:numel(colnames) 
    [ContextModel, A_keep] = fastcormics_RNAseq(Cmodel, discretized(:,i), ...
        rownames, dico, biomass_rxn, already_mapped_tag, consensus_proportion, epsilon, optional_settings);
     
    filePath = fullfile('PBMC_Models', colnames{i});
    save(filePath, 'ContextModel');   
    models_keep(A_keep,i) = 1;
end
%% 
% 

delete clone*.log
##### SOURCE END #####
--></body></html>