% This LaTeX was auto-generated from MATLAB code.
% To make changes, update the MATLAB code and export to LaTeX again.

\documentclass{article}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{lmodern}
\usepackage{graphicx}
\usepackage{color}
\usepackage{listings}
\usepackage{hyperref}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{epstopdf}
\usepackage[table]{xcolor}
\usepackage{matlab}

\sloppy
\epstopdfsetup{outdir=./}
\graphicspath{ {./CMR_rFASTCORMICS_PBMC_images/} }

\begin{document}

\matlabtitle{rFASTCORMICS FOR THE PBMC 3K scRNA-seq DATASET}

\begin{par}
\begin{flushleft}
(Pacheco et al., 2019, EBioMedicine)
\end{flushleft}
\end{par}


\vspace{1em}

\matlabheading{SET UP}

\begin{matlabcode}
addpath(genpath(pwd)) 
addpath(genpath('/Users/claudiavicentecomorera/cobratoolbox')) 
feature astheightlimit 2000
\end{matlabcode}
\begin{matlaboutput}
ans = 'feature astHeightLimit 500'
\end{matlaboutput}
\begin{matlabcode}
initCobraToolbox(false)
\end{matlabcode}
\begin{matlaboutput}
      _____   _____   _____   _____     _____     |
     /  ___| /  _  \ |  _  \ |  _  \   / ___ \    |   COnstraint-Based Reconstruction and Analysis
     | |     | | | | | |_| | | |_| |  | |___| |   |   The COBRA Toolbox - 2024
     | |     | | | | |  _  { |  _  /  |  ___  |   |
     | |___  | |_| | | |_| | | | \ \  | |   | |   |   Documentation:
     \_____| \_____/ |_____/ |_|  \_\ |_|   |_|   |   http://opencobra.github.io/cobratoolbox
                                                  | 

 > Checking if git is installed ...  Done (version: 2.39.3).
 > Checking if the repository is tracked using git ...  Done.
 > Checking if curl is installed ...  Done.
 > Checking if remote can be reached ...  Done.
 > Initializing and updating submodules (this may take a while)... Done.
 > Adding all the files of The COBRA Toolbox ...  Done.
 > Define CB map output... set to svg.
 > TranslateSBML is installed and working properly.
 > Configuring solver environment variables ...
   - [*---] ILOG_CPLEX_PATH: /Applications/CPLEX_Studio1210/cplex/matlab/x86-64_osx
   - [----] GUROBI_PATH: --> set this path manually after installing the solver ( see instructions )
   - [----] TOMLAB_PATH: --> set this path manually after installing the solver ( see instructions )
   - [----] MOSEK_PATH: --> set this path manually after installing the solver ( see instructions )
   Done.
 > Checking available solvers and solver interfaces ...Gurobi installed at this location? 
Licence file current? 
Version identifier: 12.10.0.0 | 2019-11-26 | 843d4de
CPXPARAM_Output_WriteLevel                       3
CPXPARAM_Output_CloneLog                         -1
Found incumbent of value 0.000000 after 0.00 sec. (0.00 ticks)

Root node processing (before b&c):
  Real time             =    0.01 sec. (0.00 ticks)
Parallel b&c, 8 threads:
  Real time             =    0.00 sec. (0.00 ticks)
  Sync time (average)   =    0.00 sec.
  Wait time (average)   =    0.00 sec.
                          ------------
Total (root+branch&cut) =    0.01 sec. (0.00 ticks)
Could not find installation of tomlab_cplex, so it cannot be tested
Original LP has 1 row, 2 columns, 1 non-zero
Objective value = 0
OPTIMAL SOLUTION FOUND BY LP PRESOLVER

 > [glpk] Primal optimality condition in solveCobraLP satisfied.Could not find installation of mosek, so it cannot be tested

   --------------------------------------------------------
   pdco.m                      Version pdco5 of 15 Jun 2018
   Primal-dual barrier method to minimize a convex function
   subject to linear constraints Ax + r = b,  bl <= x <= bu
                                                           
   Michael Saunders       SOL and ICME, Stanford University
   Contributors:     Byunggyoo Kim (SOL), Chris Maes (ICME)
                     Santiago Akle (ICME), Matt Zahr (ICME)
                     Aekaansh Verma (ME)                   
   --------------------------------------------------------

The objective is linear
The matrix A is an explicit sparse matrix

m        =        1     n        =        2      nnz(A)  =        1
max |b | =        0     max |x0| =  1.0e+00      xsize   =  1.0e+00
max |y0| =        1     max |z0| =  1.0e+00      zsize   =  1.0e+00

x0min    =        1     featol   =  1.0e-06      d1max   =  1.0e-04
z0min    =        1     opttol   =  1.0e-06      d2max   =  5.0e-04
mu0      =  1.0e-01     steptol  =     0.99     bigcenter=     1000

LSMR/MINRES:
atol1    =  1.0e-10     atol2    =  1.0e-15      btol    =  0.0e+00
conlim   =  1.0e+12     itnlim   =       10      show    =        0

Method   =        2     (1 or 11=chol  2 or 12=QR  3 or 13=LSMR  4 or 14=MINRES 21=SQD(LU)  22=SQD(MA57))
Eliminating dy before dx  
 

Bounds:
  [0,inf]  [-inf,0]  Finite bl  Finite bu  Two bnds   Fixed    Free
        0         0          0          0         0       2       0
  [0, bu]  [bl,  0]  excluding fixed variables
        0         0

Itn   mu stepx stepz  Pinf  Dinf  Cinf   Objective    nf  center       QR
  0                   -6.6 -99.0  -Inf  1.2500000e-07        1.0
  1 -1.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0        1
  2 -3.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
  3 -5.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
  4 -7.0 1.000 1.000 -99.0 -99.0  -Inf  0.0000000e+00  1     1.0
   Converged

max |x| =     0.000    max |y| =     0.000    max |z| =     0.000   scaled
max |x| =     0.000    max |y| =     0.000    max |z| =     0.000 unscaled
max |x| and max |z| exclude fixed variables
PDitns  =         4     QRitns =         0    cputime =       0.2

Distribution of vector     x         z
[      1,     10 )         0         2
[    0.1,      1 )         0         0
[   0.01,    0.1 )         0         0
[  0.001,   0.01 )         0         0
[ 0.0001,  0.001 )         0         0
[  1e-05, 0.0001 )         0         0
[  1e-06,  1e-05 )         0         0
[  1e-07,  1e-06 )         0         0
[  1e-08,  1e-07 )         0         0
[      0,  1e-08 )         2         0 
Elapsed time is 0.222692 seconds.

 > [pdco] Primal optimality condition in solveCobraLP satisfied.
 > [pdco] Dual optimality condition in solveCobraLP satisfied.
Elapsed time is 0.003358 seconds.
In 0.0068551 sec, file written to /Users/claudiavicentecomorera/cobratoolbox/binary/maci64/bin/minos/data/FBA/qFBA.txt
Could not find installation of cplex_direct, so it cannot be tested

 > [cplexlp] Primal optimality condition in solveCobraLP satisfied.
 > [cplexlp] Dual optimality condition in solveCobraLP satisfied.
Could not find installation of tomlab_snopt, so it cannot be tested
 Done.
 > Setting default solvers ...Could not find installation of mosek, so it cannot be tested
 Done.
 > Saving the MATLAB path ... Done.
   - The MATLAB path was saved in the default location.

 > Summary of available solvers and solver interfaces

			Support 	   LP 	 MILP 	   QP 	 MIQP 	  NLP 	   EP
	------------------------------------------------------------------------------
	gurobi       	active        	    0 	    0 	    0 	    0 	    - 	    -
	ibm_cplex    	active        	    1 	    1 	    1 	    1 	    - 	    -
	tomlab_cplex 	active        	    0 	    0 	    0 	    0 	    - 	    -
	glpk         	active        	    1 	    1 	    - 	    - 	    - 	    -
	mosek        	active        	    0 	    - 	    0 	    - 	    - 	    0
	matlab       	active        	    1 	    - 	    - 	    - 	    1 	    -
	pdco         	active        	    1 	    - 	    1 	    - 	    - 	    1
	quadMinos    	active        	    1 	    - 	    - 	    - 	    - 	    -
	dqqMinos     	active        	    1 	    - 	    1 	    - 	    - 	    -
	cplex_direct 	active        	    0 	    0 	    0 	    - 	    - 	    -
	cplexlp      	active        	    1 	    - 	    - 	    - 	    - 	    -
	qpng         	passive       	    - 	    - 	    1 	    - 	    - 	    -
	tomlab_snopt 	passive       	    - 	    - 	    - 	    - 	    0 	    -
	lp_solve     	legacy        	    1 	    - 	    - 	    - 	    - 	    -
	------------------------------------------------------------------------------
	Total        	-             	    8 	    2 	    4 	    1 	    1 	    1

 + Legend: - = not applicable, 0 = solver not compatible or not installed, 1 = solver installed.


 > You can solve LP problems using: 'ibm_cplex' - 'glpk' - 'pdco' - 'cplexlp' 
 > You can solve MILP problems using: 'ibm_cplex' - 'glpk' 
 > You can solve QP problems using: 'ibm_cplex' - 'pdco' 
 > You can solve MIQP problems using: 'ibm_cplex' 
 > You can solve NLP problems using: 
 > You can solve EP problems using: 'pdco' 

Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
Gurobi installed at this location? 
Licence file current? 
> Checking for available updates ... skipped
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/componentContribution/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/groupContribution/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/inchi/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/molFiles/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/protons/new
removing: /Users/claudiavicentecomorera/cobratoolbox/src/analysis/thermo/trainingModel/new
\end{matlaboutput}

\begin{par}
\begin{flushleft}
Import the data and split into 3 variables:
\end{flushleft}
\end{par}

\begin{matlabcode}
data = readtable('data_pbmc.csv');
\end{matlabcode}
\begin{matlaboutput}
Warning: Column headers from the file were modified to make them valid MATLAB identifiers before creating variable names for the table. The original column headers are saved in the VariableDescriptions property.
Set 'PreserveVariableNames' to true to use the original column headers as table variable names.
\end{matlaboutput}
\begin{matlabcode}
colnames = data.Properties.VariableNames(2:end);
rownames = table2cell(data(:, 1));
CPM = table2array(data(:, 2:end))
\end{matlabcode}
\begin{matlaboutput}
CPM = 32738x2638    
1.0e+05 *

         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0
         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0         0

\end{matlaboutput}


\vspace{1em}

\vspace{1em}
\matlabheading{DISCRETIZATION STEP}

\begin{par}
\begin{flushleft}
Discretize the gene expression values into expressed, not expressed and unknown expression status:
\end{flushleft}
\end{par}

\begin{matlabcode}
discretized = discretize_FPKM(CPM, colnames) 
\end{matlabcode}
\begin{matlaboutput}
discretized = 32738x2638    
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1
    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1    -1

\end{matlaboutput}


\vspace{1em}

\matlabheading{rFASTCORMICS}

\matlabheadingtwo{NEEDED INPUTS}

\begin{par}
\begin{flushleft}
Apart from the data, rFASTCORMICS needs some more inputs before run:
\end{flushleft}
\end{par}

\begin{itemize}
\setlength{\itemsep}{-1ex}
   \item{\begin{flushleft} A file which links the geneIDs from the RNA-seq data to the gene identifiers in the model (Entrez). \end{flushleft}}
   \item{\begin{flushleft} A consistent reconstruction (as Recon3D) \end{flushleft}}
\end{itemize}

\begin{matlabcode}
load dico_rFASTCORMICS.mat 
load Recon3DModel_301.mat;
A = fastcc_4_rfastcormics(model, 1e-4,0) 
\end{matlabcode}
\begin{matlaboutput}
The input model is consistent.
Elapsed time is 14.364739 seconds.
A = 10600x1    
     1
     2
     3
     4
     5
     6
     7
     8
     9
    10

\end{matlaboutput}
\begin{matlabcode}
Cmodel = removeRxns(model, model.rxns(setdiff(1:numel(model.rxns),A)))
\end{matlabcode}
\begin{matlaboutput}
Cmodel = 
                      S: [5835x10600 double]
                   mets: {5835x1 cell}
                      b: [5835x1 double]
                 csense: [5835x1 char]
                   rxns: {10600x1 cell}
                     lb: [10600x1 double]
                     ub: [10600x1 double]
                      c: [10600x1 double]
                 osense: -1
                  genes: {2248x1 cell}
                  rules: {10600x1 cell}
             metCharges: [5835x1 int64]
            metFormulas: {5835x1 cell}
              metSmiles: {5835x1 cell}
               metNames: {5835x1 cell}
              metHMDBID: {5835x1 cell}
         metInChIString: {5835x1 cell}
              metKEGGID: {5835x1 cell}
           metPubChemID: {5835x1 cell}
            description: 'Recon3DModel.mat'
                grRules: {10600x1 cell}
             rxnGeneMat: [10600x2248 double]
    rxnConfidenceScores: [10600x1 double]
               rxnNames: {10600x1 cell}
               rxnNotes: {10600x1 cell}
           rxnECNumbers: {10600x1 cell}
          rxnReferences: {10600x1 cell}
              rxnKEGGID: {10600x1 cell}
             subSystems: {10600x1 cell}
             metCHEBIID: {5835x1 cell}
               metPdMap: {5835x1 cell}
            metReconMap: {5835x1 cell}
                modelID: 'Recon3DModel'
                 rxnCOG: {10600x1 cell}
       rxnKeggOrthology: {10600x1 cell}
            rxnReconMap: {10600x1 cell}
                version: 'Recon3D_01'
             PleaseCite: 'Brunk et al, Nat Biotech, 2018; doi:10.1038/nbt.4072'

\end{matlaboutput}


\vspace{1em}

\vspace{1em}
\matlabheadingtwo{PARAMETERS AND OPTIONAL INPUTS}

\begin{par}
\begin{flushleft}
We will use the parameters and inputs defined in rFASTCORMICS example scripts, excluding the medium constraint.
\end{flushleft}
\end{par}

\begin{matlabcode}
biomass_rxn = {'biomass_maintenance'}; 
already_mapped_tag = 0;
epsilon = 1e-4; 
consensus_proportion = 0.9; 
unpenalizedSystems = {'Transport, endoplasmic reticular';
    'Transport, extracellular';
    'Transport, golgi apparatus';
    'Transport, mitochondrial';
    'Transport, peroxisomal';
    'Transport, lysosomal';
    'Transport, nuclear'};
Cmodel_subSystems = cellfun(@(x) x{1}, Cmodel.subSystems, 'UniformOutput', false); 
unpenalized = Cmodel.rxns(ismember(Cmodel_subSystems,unpenalizedSystems));

optional_settings.unpenalized = unpenalized; 
optional_settings.func = {'biomass_maintenance','DM_atp_c_'}; 
not_medium_constrained = 'EX_tag_hs[e]';
optional_settings.not_medium_constrained = not_medium_constrained;
\end{matlabcode}


\vspace{1em}

\matlabheadingtwo{CONTEXT-SPECIFIC SINGLE MODELS CREATION}

\begin{par}
\begin{flushleft}
By setting all the inputs into the \textit{\textbf{fastcormics\_RNAseq}} function, a model for each cell (column) of the data will be build:
\end{flushleft}
\end{par}

\begin{matlabcode}
for i = 1:numel(colnames) 
    [ContextModel, A_keep] = fastcormics_RNAseq(Cmodel, discretized(:,i), ...
        rownames, dico, biomass_rxn, already_mapped_tag, consensus_proportion, epsilon, optional_settings);
     
    filePath = fullfile('PBMC_Models', colnames{i});
    save(filePath, 'ContextModel');   
    models_keep(A_keep,i) = 1;
end
\end{matlabcode}
\begin{matlaboutput}
creating model.rev
unnesting subsystems
2143 of 2248 genes matched
Warning: No optional settings detected
Elapsed time is 46.868621 seconds.
Error using save
Cannot create 'AAACATACAACCAC_1.mat' because 'PBMC_Models' does not exist.
\end{matlaboutput}


\vspace{1em}
\begin{matlabcode}
delete clone*.log
\end{matlabcode}

\end{document}
